### PR TITLE
Fix run_tests verbosity

### LINKED ISSUE #253: `catkin_test_results --verbose` explodes travis log beyond max
with the changes introduced in #251 (related to #118), our Travis jobs began to fail due to exceeding log length maximum (we haven't seen this earlier due to other Travis/Docker issues - see #247 #248)

```
The log length has exceeded the limit of 4 MB (this usually means that the test suite is raising the same exception over and over).

The job has been terminated
```

I can ensure that "the test suite is raising the same exception over and over" is **not** true in our case!

To give some numbers:
 - all CI steps from start to build-step produce about 2000 lines of output
 - the test-step (--verbose) produces more than 8000 lines of output....

Is there a way to fix #118 without using `--verbose` during the test-step?

Our `travis.yml` looks pretty standard (and has worked before):
```
sudo: required
dist: trusty
language: generic
notifications:
  email:
    on_success: change
    on_failure: always
env:
  global:
    - ROS_REPO=ros
    - NOT_TEST_INSTALL=true
  matrix:
    - ROS_DISTRO=indigo UPSTREAM_WORKSPACE=file
    - ROS_DISTRO=indigo UPSTREAM_WORKSPACE=file $ROSINSTALL_FILENAME="custom.rosinstall"
    - ROS_DISTRO=kinetic UPSTREAM_WORKSPACE=file
matrix:
  allow_failures:
    - env: ROS_DISTRO=kinetic UPSTREAM_WORKSPACE=file
install:
  - git clone https://github.com/ros-industrial/industrial_ci.git .ci_config
script:
  - .ci_config/travis.sh
#  - ./travis.sh  # Enable this when you have a package-local script 
```

### PR DESCRIPTION
**Pull request by ipa-mdl**
This is my attempt to fix #253, as a replacement for #255 

It reverts #244, but sets `TARGET_PKGS` to all packages to address #232.
To make sure that all downstream packages and their dependencies get built, a dedicated step was added.

In addition `tests` will be built before `run_tests` to build the tests with the build options. 

@miguelprada: please test your bare rosinstall usecase
@ipa-fxm: please test with your setup

### CODE REVIEW

**Comment by miguelprada** [comment]
The bare rosinstall usecase still works. Thanks for keeping it in mind, @ipa-mdl!

**Comment by ipa-mdl** [comment]
@130s: do you have any objections?

**Comment by ipa-mdl** [comment]
`TARGET_PKGS` won't get touched if it was provided by the user.

Old behavior:
Fill it with the the names of the packages in target repo.
For the special case of #232, it was empty.
For `catkin build` this was okay, because it just build all packages.
But `catin run_tests` because `--no-deps` requires listing the package names.

New behavior:
1. Fill it with the the names of the packages in target repo.
2. If this empty, fill it with all package names.

So `catkin build` will still build all, but in addition `catkin run_tests`  will run all tests.

The only side effect I can think of is that users with empty target repos cannot blacklist packages in `CATKIN_CONFIG` anymore (@miguelprada: so, beware! ^^).
But this is true for regular users as well (with or without this PR).
A proper solution would be #172

File: industrial_ci/src/tests/source_tests.sh [hunk 1]
```diff
@@ -163,16 +163,29 @@ ici_time_start catkin_build
 
 # for catkin
 if [ "${TARGET_PKGS// }" == "" ]; then export TARGET_PKGS=`catkin_topological_order ${TARGET_REPO_PATH} --only-names`; fi
+# fall-back to all packages
+if [ "${TARGET_PKGS// }" == "" ]; then export TARGET_PKGS=`catkin_topological_order $CATKIN_WORKSPACE/src --only-names`; fi
```
Comment (130s) [inline industrial_ci/src/tests/source_tests.sh]: So this line and previous line is the "new behavior (regarding `TARGET_PKGS`)" in https://github.com/ros-industrial/industrial_ci/pull/256#issuecomment-356976994 . I'd never figure out by myself why these very similar 2 lines are needed. For future maintenance, would you add a comment? Just add a link is fine IMO.

Comment (ipa-mdl) [reply industrial_ci/src/tests/source_tests.sh]: Addressed.

**Pull request merged**